File: client/src/app/models/helpers/member-params.ts
import {PaginationParams} from "./paginationParams";

export class MemberParams extends PaginationParams {
  userNameOrKnownAs: string | undefined;
  gender: string | undefined;
  minAge = 18;
  maxAge = 99;
  orderBy = 'lastActive';

  constructor(gender: string) {
    super(); // Constructors of derived classes must contain a 'super' call to properly set up the inheritance chain.
    this.gender = gender === 'female' ? 'male' : 'female';
  }
}

File: backend/Da.Application/DTOs/Parameters/MemberParams.cs
namespace Da.Application.DTOs.Parameters;

public class MemberParams : PaginationParams
{
    public string? UserId { get; set; }
    public string? UserNameOrKnownAs { get; set; }
    public string? Gender { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActive";
}